pygeos/strtree.py

```diff
@@ -186,3 +189,126 @@ def query_bulk(self, geometry, predicate=None):
             predicate = BinaryPredicate[predicate].value
 
         return self._tree.query_bulk(geometry, predicate)
+
+    @requires_geos("3.6.0")
+    def nearest(self, geometry):
+        """Returns the index of the nearest item in the tree for each input
+        geometry.
+
+        If there are multiple equidistant or intersected geometries in the tree,
+        only a single result is returned for each input geometry, based on the
+        order that tree geometries are visited; this order may be
+        nondeterministic.
+
+        Any geometry that is None or empty in the input geometries is omitted
+        from the output.
+
+        Parameters
+        ----------
+        geometry : Geometry or array_like
+            Input geometries to query the tree.
+
+        Returns
+        -------
+        ndarray with shape (2, n)
```

Comment:
Since we drop input geometries that are empty or missing, we needed a way to specify which input geometries had corresponding nearest geometries in the result.

Reply:
I see, integers do not have a NaN equivalent so this is OK like this.